### PR TITLE
Fix Swap wasp.png to wasp.svg

### LINKED ISSUE #1534: Replace the `png` Wasp logo with an `svg` in default template
With every new Wasp project we put a `png` Wasp logo which is `25kB` and the equivalent `svg` file is only `4kB`.

Also, the SVG logo works on every resolution and that's kinda the standard nowadays. Small improvement, but it's a cool one.

### Implementation
The logo lives in the `waspc/data/Cli/templates/basic/src/client/waspLogo.png`

The change should:
1. Swap out the PNG logo with a SVG one (ideally the SVG one would be a React component https://react-svgr.com/playground/)
2. Test out that the new template works as expected
3. Fix e2e tests

### PR DESCRIPTION
**Pull request by lalitkumawat1m**
### Description

Fixes #1534

### Select what type of change this PR introduces:

1. [ ] **Just code/docs improvement** (no functional change). 
2. [x] **Bug fix** (non-breaking change which fixes an issue).
3. [ ] **New feature** (non-breaking change which adds functionality).
4. [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

### Update Waspc ChangeLog and version if needed
If you did a bug fix, new feature, or breaking change, that affects waspc, make sure you satisfy the following:
1. [ ] I updated [ChangeLog.md](https://github.com/wasp-lang/wasp/blob/main/waspc/ChangeLog.md) with description of the change this PR introduces.
2. [ ] I bumped waspc version in [waspc.cabal](https://github.com/wasp-lang/wasp/blob/main/waspc/waspc.cabal) to reflect changes I introduced, with regards to the version of the latest wasp release, if the bump was needed.

### CODE REVIEW

**Comment by infomiho** [comment]
@lalitkumawat1m did you test out the templates changes locally with `wasp new`?

**Comment by infomiho** [comment]
If you tested it out locally, why does the CI fail on the e2e tests? 🙃

**Comment by Martinsos** [comment]
> If you tested it out locally, why does the CI fail on the e2e tests? 🙃

It seems to me that e2e tests merely need updating, that that is the cause of them failing? That there is no actual error? This comes from me looking at output of a CI action that failed.

File: waspc/ChangeLog.md [hunk 1]
```diff
@@ -903,7 +903,7 @@ This was the file tree of a newly generated project in the previous version of W
 │   ├── Main.css
 │   ├── MainPage.js
 │   ├── .waspignore
-│   └── waspLogo.png
+│   └── waspLogo.svg
```
Comment (Martinsos) [inline waspc/ChangeLog.md:906]: This shouldn't be changed, this and the line below, since this is ChangeLog -> so this is how it was at that point in the time and should be kept that way.

File: web/blog/2022-11-29-typescript-feature-announcement.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ This will generate a project skeleton in the folder `myApp`. The project structu
 │   │   ├── MainPage.jsx
 │   │   ├── react-app-env.d.ts
 │   │   ├── tsconfig.json
-│   │   └── waspLogo.png
+│   │   └── waspLogo.svg
```
Comment (Martinsos) [inline web/blog/2022-11-29-typescript-feature-announcement.md:49]: Old blog post, so let's keep it as it was.

File: web/blog/2023-06-27-build-your-own-twitter-agent-langchain.md [hunk 3]
```diff
@@ -1104,7 +1104,7 @@ Ok, but what good is a function that fetches the data if we’ve got nowhere to
 Let’s go now to our `src/client/MainPage.tsx` file (make sure it’s got the `.tsx` extension and not `.jsx`) and replace the contents with these below:
 
 ```tsx
-import waspLogo from './waspLogo.png'
+import waspLogo from './waspLogo.svg'
```
Comment (Martinsos) [inline web/blog/2023-06-27-build-your-own-twitter-agent-langchain.md:1107]: Also old blog post, let's keep it as it was.

File: waspc/e2e-test/test-outputs/waspBuild-golden/files.manifest [hunk 4]
```diff
@@ -52,7 +52,7 @@ waspBuild/.wasp/build/web-app/src/ext-src/Main.css
 waspBuild/.wasp/build/web-app/src/ext-src/MainPage.jsx
 waspBuild/.wasp/build/web-app/src/ext-src/vite-env.d.ts
 waspBuild/.wasp/build/web-app/src/ext-src/vite.config.ts
-waspBuild/.wasp/build/web-app/src/ext-src/waspLogo.png
+waspBuild/.wasp/build/web-app/src/ext-src/waspLogo.svg
```
Comment (Martinsos) [inline waspc/e2e-test/test-outputs/waspBuild-golden/files.manifest:55]: You did search and replace here, but that is not how our e2e tests work. To update them correctly, you actually need to build and run the waspc Haskell project on your machine and update the e2e tests. Which is involved, you need to have Haskell Toolchain running on your machine (GHC, Cabal). We could run these for you possibly though, if you get everything else right! But you should certainly revert these manual changes on e2e tests that you did (everything inside of e2e-test dir).

File: waspc/data/Cli/templates/basic/src/client/MainPage.jsx [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import waspLogo from './waspLogo.png'
+import waspLogo from './waspLogo.svg'
```
Comment (Martinsos) [inline waspc/data/Cli/templates/basic/src/client/MainPage.jsx:1]: This is really one of the most important changes, this change here.
This tells Wasp to, when it generates new project, use .svg in it.
But that waspLogo.svg, you need to provide it also somewhere in the `data/Cli/templates/basic/src/client/` dir, I don't think you did though, I don't think you replaced that waspLogo.png with waspLogo.svg? You should do that.

File: examples/cronjobs-analytics-waspleau/src/client/WaspSourceHeader.jsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import waspLogo from './waspLogo.png'
+import waspLogo from './waspLogo.svg'
```
Comment (Martinsos) [inline examples/cronjobs-analytics-waspleau/src/client/WaspSourceHeader.jsx:1]: All these changes you did in examples -> they are not really needed. It doesn't really matter if they use waspLogo.png, it is just a file generated at start. So I would revert these changes.

Comment (Martinsos) [reply examples/cronjobs-analytics-waspleau/src/client/WaspSourceHeader.jsx:1]: Also, most of the example you corrected only half way -> you changed the import, but didn't swap .png file with .svg file. You did that in only one example. So I believe only one example works correctly.

**Review by Martinsos — CHANGES_REQUESTED** [review]
@lalitkumawat1m Thanks for the effort!

I did a thorough review now and this PR is not in a great shape.
You mostly did global waspLogo.png -> waspLogo.svg string replace -> a lot of those replacements are incorrect or unneeded.
Also, you didn't follow those up with replacing actual waspLogo.png files with waspLogo.svg files except in one place.

In this state, I doubt that any but one example is correctly showing its logo, and all new projects that wasp would generate with `wasp new` I believe are also broken.

So I am not sure what you tested, when you said you tested it?

I left comments that will hopefully help with getting this PR to a better shape, but testing and understanding the changes is also a part of doing PR, so if you want to continue with this PR please put effort into that.

**Comment by infomiho** [comment]
Thank you for the contribution but this change affects our starter templates which are crucial for our Wasp CLI.

I will now close this PR as I'm not confident that we tested out the changes in a good way. If you manage to test it locally, feel free to reopen.
